File: Forms/Frm_Main.cs
using System;
using System.Speech.AudioFormat;
using System.Speech.Synthesis;
using System.Windows.Forms;

using SpeakingClock.Classes;

namespace SpeakingClock.Forms
{
    public partial class Frm_Main : Form
    {
        public Frm_Main()
        {
            InitializeComponent();
        }

        private void Frm_Main_Load(object sender, EventArgs e)
        {
            this.Text = "Speaking Clock v" + System.Reflection.Assembly.GetExecutingAssembly().GetName().Version.ToString();

            Tim_Main.Enabled = true;
        }

        private void Frm_Main_FormClosing(object sender, FormClosingEventArgs e)
        {
            GenericFunctions.CloseForm(e);
        }

        private void Tim_Main_Tick(object sender, EventArgs e)
        {
            Lb_RealTime.Text = DateTime.Now.ToString("yyyy-MM-dd hh:mm:ss.ff tt", GlobalVariables.cul_invar);
        }

        private void Ch_AnnounceEvery_CheckedChanged(object sender, EventArgs e)
        {
            Num_AnnounceEvery.Enabled = Ch_AnnounceEvery.Checked;
        }

        private void Lb_Speak_Click(object sender, EventArgs e)
        {
            if (Bg_Speak.IsBusy) return;
            Lb_Speak.Enabled = false;
            Bg_Speak.RunWorkerAsync();
        }

        private void Bg_Speak_DoWork(object sender, System.ComponentModel.DoWorkEventArgs e)
        {
            using (SpeechSynthesizer synth = new SpeechSynthesizer())
            {
                synth.SetOutputToDefaultAudioDevice();
                synth.SelectVoice("Microsoft Zira Desktop");
                synth.Speak("It's now " + DateTime.Now.ToString("hh:mm tt", GlobalVariables.cul_invar));
            }
        }

        private void Bg_Speak_RunWorkerCompleted(object sender, System.ComponentModel.RunWorkerCompletedEventArgs e)
        {
            Lb_Speak.Enabled = true;

            using (SpeechSynthesizer synth = new SpeechSynthesizer())
            {
                // Output information about all of the installed voices.
                Console.WriteLine("Installed voices -");
                foreach (InstalledVoice voice in synth.GetInstalledVoices())
                {
                    VoiceInfo info = voice.VoiceInfo;
                    string AudioFormats = "";
                    foreach (SpeechAudioFormatInfo fmt in info.SupportedAudioFormats)
                    {
                        AudioFormats += String.Format("{0}\n",
                        fmt.EncodingFormat.ToString());
                    }

                    Console.WriteLine(" Name:          " + info.Name);
                    Console.WriteLine(" Culture:       " + info.Culture);
                    Console.WriteLine(" Age:           " + info.Age);
                    Console.WriteLine(" Gender:        " + info.Gender);
                    Console.WriteLine(" Description:   " + info.Description);
                    Console.WriteLine(" ID:            " + info.Id);
                    Console.WriteLine(" Enabled:       " + voice.Enabled);
                    if (info.SupportedAudioFormats.Count != 0)
                    {
                        Console.WriteLine(" Audio formats: " + AudioFormats);
                    }
                    else
                    {
                        Console.WriteLine(" No supported audio formats found");
                    }

                    string AdditionalInfo = "";
                    foreach (string key in info.AdditionalInfo.Keys)
                    {
                        AdditionalInfo += String.Format("  {0}: {1}\n", key, info.AdditionalInfo[key]);
                    }

                    Console.WriteLine(" Additional Info - " + AdditionalInfo);
                    Console.WriteLine();
                }
            }
        }
    }
}

File: Forms/Frm_Main.vb
Public Class Frm_Main

#Region "Form"

    Private Sub Frm_Main_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Sub_FormIconText(Me, "Speaking Clock")
        Not_Main.Icon = My.Resources.art

        Tim_MainTimer.Enabled = True
    End Sub

    Private Sub Frm_Main_FormClosing(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing
        'e.Cancel = True
        'Me.Hide()
    End Sub

#End Region

#Region "Controls"

    Private Sub Tim_MainTimer_Tick(sender As Object, e As EventArgs) Handles Tim_MainTimer.Tick
        Dim nowTimeDate As DateTime = Now
        Lb_Time.Text = nowTimeDate.ToString("hh:mm:ss tt", cul_invar)
        Lb_Date.Text = nowTimeDate.ToString("ddd, yyyy-MMM-dd", cul_invar)
        Not_Main.Text = nowTimeDate.ToString("hh:mm:ss tt" & str_newLine & "ddd, yyyy-MMM-dd", cul_invar)
    End Sub

    Private Sub Lb_Settings_Click(sender As Object, e As EventArgs) Handles Lb_Settings.Click
        With Frm_Settings
            .ShowDialog(Me)
        End With
    End Sub

    Private Sub Tim_Announce_Tick(sender As Object, e As EventArgs) Handles Tim_Announce.Tick

    End Sub


#End Region

End Class
